Updates to README.md, and initial CONTRIBUTING.md, GOVERNANCE.md, INSTALL.md

cary-ilm opened the pull request:
All work in progress, cobbled together from OpenColorIO, OpenVDB, OpenCue and other well-documented projects.

cary-ilm committed:
Meeting notes 2019-5-2

cary-ilm committed:
formatting

seabeepea committed:
added GOVERNANCE.md

seabeepea committed:
CONTRIBUTING.md, INSTALL.md, and changes README.md and INSTALL.md

seabeepea committed:
Merge https://github.com/openexr/openexr

seabeepea committed:
Contributing and Goverance sections

seabeepea committed:
typos

cary-ilm committed:
Edits to README.md and CONTRIBUTING.md

xlietz committed:
fixing minor typos

lgritz commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,409 @@
+Every commit must be signed off.  That is, every commit log message
+must include a “`Signed-off-by`” line (generated, for example, with
+“`git commit --signoff`”), indicating that the committer wrote the
+code and has the right to release it under the [MPL
```
OpenEXR is not MPL, right?

lgritz commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,409 @@
+They look like this:
+
+    /// Explanation of a class.  Note THREE slashes!
+    /// Also, you need at least two lines like this.  If you don't have enough
```
This is no longer true -- I don't know if I misunderstood originally, or if Doxygen has gotten smarter over the years. But the blank commented line is not necessary and Doxygen is perfectly happy with a one-line triple-slash comment.

cary-ilm commented:
Yep - copy/paste error, I think that came from OpenVDB.

On Wed, Jun 12, 2019 at 11:30 AM Larry Gritz <notifications@github.com>
wrote:

> *@lgritz* commented on this pull request.
> ------------------------------
>
> In CONTRIBUTING.md
> <https://github.com/openexr/openexr/pull/397#discussion_r293057109>:
>
> > +Licence Agreement](http://www.openexr.com/downloads/OpenEXRCorporateContributorLicenseAgreement.docx).
>
> +
>
> +Download the appropriate CLA from the links above (or find them in the
>
> +src/doc directory of the software distribution), print, sign, and
>
> +rescan it (or just add a digital signature directly), and email it
>
> +back to us at [info@openexr.com](info@openexr.com).
>
> +
>
> +The OpenEXR CLA's are the standard forms used by Linux Foundation
>
> +projects.
>
> +
>
> +### Commit Sign-Off
>
> +
>
> +Every commit must be signed off.  That is, every commit log message
>
> +must include a “`Signed-off-by`” line (generated, for example, with
>
> +“`git commit --signoff`”), indicating that the committer wrote the
>
> +code and has the right to release it under the [MPL
>
>
> OpenEXR is not MPL, right?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openexr/openexr/pull/397?email_source=notifications&email_token=AFC3DGKTGO67MAIPGAOQF6TP2E6EBA5CNFSM4HXNNXY2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB3LI5KA#pullrequestreview-248942248>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFC3DGO45ND2ATFJ6XKGNADP2E6EBANCNFSM4HXNNXYQ>
> .
>


-- 
Cary Phillips | R&D Supervisor | ILM | San Francisco

cary-ilm commented:
I lifted the code style section from OCIO and haven't modified it yet.

On Wed, Jun 12, 2019 at 11:33 AM Larry Gritz <notifications@github.com>
wrote:

> *@lgritz* commented on this pull request.
> ------------------------------
>
> In CONTRIBUTING.md
> <https://github.com/openexr/openexr/pull/397#discussion_r293058390>:
>
> > +Feel free to use Boost classes you already see in the code base, but don't
> +use any Boost you don't see us already using, without first checking with
> +the project leader.
> +
> +#### Comments and Doxygen
> +
> +Comment philosophy: try to be clear, try to help teach the reader what's
> +going on in your code.
> +
> +Prefer C++ comments (starting line with `//`) rather than C comments (`/* ... */`).
> +
> +For public APIs we tend to use Doxygen-style comments (start with `///`).
> +They look like this:
> +
> +    /// Explanation of a class.  Note THREE slashes!
> +    /// Also, you need at least two lines like this.  If you don't have enough
>
> This is no longer true -- I don't know if I misunderstood originally, or
> if Doxygen has gotten smarter over the years. But the blank commented line
> is not necessary and Doxygen is perfectly happy with a one-line
> triple-slash comment.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openexr/openexr/pull/397?email_source=notifications&email_token=AFC3DGIQ7QN4IJLJIZLNFQDP2E6PBA5CNFSM4HXNNXY2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB3LJJXA#pullrequestreview-248943836>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFC3DGPX5UG53W4SL76NMSLP2E6PBANCNFSM4HXNNXYQ>
> .
>


-- 
Cary Phillips | R&D Supervisor | ILM | San Francisco

lgritz commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,409 @@
+
+### How to Ask for Help
+
+If you have trouble installing, building, or using the library, but there's not yet reason to suspect you've encountered a genuine bug, start by posting a question to the [openexr-dev](http://lists.aswf.io/openexr-dev) mailing list. This is the place for question such has "How do I...".
```
I love this paragraph and am going to steal it for my projects.

cary-ilm commented:
I thought I stole that from you.... ;)

On Wed, Jun 12, 2019 at 12:46 PM Larry Gritz <notifications@github.com>
wrote:

> *@lgritz* commented on this pull request.
> ------------------------------
>
> In CONTRIBUTING.md
> <https://github.com/openexr/openexr/pull/397#discussion_r293086973>:
>
> > +explains our contribution process and procedures
> +
> +## Getting Information
> +
> +There are two primary ways to connect with the OpenEXR project:
> +
> +* The [openexr-dev](https://lists.aswf.io/g/openexr-dev) mail list:
> +This is a development focused mail list with a deep history of
> +technical conversations and decisions that have shaped the project.
> +
> +* [GitHub Issues](https://github.com/openexr/openexr/issues): GitHub
> +**issues** are used both to track bugs and to discuss feature requests.
> +
> +### How to Ask for Help
> +
> +If you have trouble installing, building, or using the library, but there's not yet reason to suspect you've encountered a genuine bug, start by posting a question to the [openexr-dev](http://lists.aswf.io/openexr-dev) mailing list. This is the place for question such has "How do I...".
>
> I love this paragraph and am going to steal it for my projects.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openexr/openexr/pull/397?email_source=notifications&email_token=AFC3DGLPH3VRGXAFKQECTN3P2FHB7A5CNFSM4HXNNXY2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB3LSBJY#pullrequestreview-248979623>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFC3DGOXQNG733F6ZWMUTBTP2FHB7ANCNFSM4HXNNXYQ>
> .
>


-- 
Cary Phillips | R&D Supervisor | ILM | San Francisco

seabeepea committed:
hello, world

seabeepea committed:
Merge branch 'master' of https://github.com/cary-ilm/openexr

cary-ilm committed:
- Formatting section is TBD
- fixed references to license
- removed references to CI
- added section on GitHub labels

Signed-off-by: Cary Phillips <cary@ilm.com>
